wagtail/query.py

```diff
@@ -608,3 +608,38 @@ def __iter__(self):
                     category=RuntimeWarning,
                 )
                 yield obj
+
+
+def first_common_ancestor(pages, model=None, include_self=False, strict=False):
+    """
+    This is similar to `PageQuerySet.first_common_ancestor` but works
+    for a list of pages instead of a queryset.
+    """
+    if model is None:
+        from wagtail.models import Page
+
+        model = Page
+
+    if not pages:
+        if strict:
+            raise model.DoesNotExist("Can not find ancestor of empty queryset")
```

Comment:
Nit: We're working on lists here, not querysets
```suggestion
            raise model.DoesNotExist("Can not find ancestor of empty list")
```